### DISCUSSION TITLE
Saving and loading sessions

### DISCUSSION THREAD

**Discussion by inclinedadarsh** [jakobhoeg/nextjs-ollama-llm-ui#56] [Q&A] [answered]
Hey, I want to know more about how does the web app is saving and loading sessions?

As far as I know, ollama doesn't save session by default. So is it using the native ollama feature to save every where session created with it or is it doing something on it's own where it's just saving the chats but not exactly the session?

Thank you!

**Comment by jakobhoeg** [comment, marked as answer]
Hey, @inclinedadarsh.

It's not using any ollama features, it's basically just saving and retrieving messages from localStorage under a generated chatId.

**Reply by inclinedadarsh** [reply]
So if I stop the nextjs server and restart it, will the model have context to previous chat messages?

**Reply by jakobhoeg** [reply]
Yes. It feeds the chathistory (all messages) of the current chat to the model whenever you send a new prompt.

**Reply by inclinedadarsh** [reply]
Please correct me if I'm wrong. The web app will basically feed the entire chat history to the model as soon I send the first message (after restarting the web app in the same chat)

**Reply by jakobhoeg** [reply]
If you're in a chat with existing messages, it will include all those in the history with the model as well as your new message. 
If you're in a new chat, only the first message will be sent (obviously).

**Reply by inclinedadarsh** [reply]
Cool. Thanks for the answer and follow ups.
